components/gitpod-db/src/typeorm/project-db-impl.ts

```diff
@@ -69,9 +69,20 @@ export class ProjectDBImpl extends TransactionalDBImpl<ProjectDB> implements Pro
         return repo.find(conditions);
     }
 
-    public async findProjects(orgId: string): Promise<Project[]> {
+    public async findProjects(orgId: string, limit?: number): Promise<Project[]> {
         const repo = await this.getRepo();
-        return repo.find({ where: { teamId: orgId, markedDeleted: false }, order: { name: "ASC" } });
+
+        const queryBuilder = repo
+            .createQueryBuilder("project")
+            .where("project.teamId = :teamId", { teamId: orgId })
+            .orderBy("project.creationTime", "DESC")
+            .andWhere("project.markedDeleted = false");
+
+        if (limit) {
```

Comment:
nit (not-blocking: Do you think we need a default value and max value (where we use it)?

Reply:
Sounds good to impose a limit here. Added that in [`1c5a88d` (#20151)](https://github.com/gitpod-io/gitpod/pull/20151/commits/1c5a88db3cf967a5af8feabc9bfd3bec63a0aa2e)